src/postgres/src/backend/utils/misc/pg_yb_utils.c

```diff
@@ -1474,3 +1476,130 @@ yb_hash_code(PG_FUNCTION_ARGS)
 	uint16_t hashed_val = YBCCompoundHash(arg_buf, total_bytes);
 	PG_RETURN_UINT16(hashed_val);
 }
+
+/*---------------------------------------------------------------------------*/
+/* Deterministic DETAIL order                                                */
+/*---------------------------------------------------------------------------*/
+
+static int yb_detail_sort_comparator(const void *a, const void *b)
+{
+	return strcmp(*(const char **) a, *(const char **) b);
+}
+
+static int default_detail_store_capacity = 10;
+
+typedef struct {
+	char ** lines;
+	int capacity;
+	int length;
+} DetailStore;
+
+void initDetailStore(DetailStore *v)
+{
+	v->capacity = default_detail_store_capacity;
+	v->lines = (char **)malloc(sizeof(char *) * v->capacity);
```

Comment:
After commit `683f313`, replacing malloc / realloc / free with palloc / repalloc and pfree works fine, so we can replace these.